src/tools/perf/cuda/cuda_alloc.c

```diff
@@ -69,15 +71,31 @@ uct_perf_cuda_alloc_reg_mem(const ucx_perf_context_t *perf,
                             unsigned flags,
                             uct_allocated_memory_t *alloc_mem)
 {
+    uct_md_attr_v2_t md_attr = {.field_mask = UCT_MD_ATTR_FIELD_REG_ALIGNMENT};
+    size_t reg_align;
+    void *reg_address;
+    size_t reg_length;
     ucs_status_t status;
 
+    status = uct_md_query_v2(perf->uct.md, &md_attr);
+    if (status != UCS_OK) {
+        ucs_error("uct_md_query_v2() returned %d", status);
+        return status;
+    }
+
     status = ucx_perf_cuda_alloc(length, mem_type, &alloc_mem->address);
     if (status != UCS_OK) {
         return status;
     }
 
-    status = uct_md_mem_reg(perf->uct.md, alloc_mem->address,
-                            length, flags, &alloc_mem->memh);
+    /* Register memory respecting MD reg_alignment */
+    reg_align   = ucs_max(md_attr.reg_alignment, 1);
+    reg_address = alloc_mem->address;
+    reg_length  = length;
```

Comment:
IMO we can remove these local helper vars

Reply:
reg_align - we can remove it if agree to init reg_alignment to 1 for all MDs
reg_address - we cannot remove it, because we don't want to modify `alloc_mem->address`
reg_length - this is safe to remove

Reply:
ok